src/entities.py

```diff
@@ -92,7 +97,305 @@ def __hash__(self) -> int:
+    MSC = "Marine Stewardship Council", "MSC", None
+    DYESTUFF = "Farbstoffe", "1", None
+    PRESERVATIVES = "Preservate", "2", None
+    ANTIOXIDANTS = "Antioxidanten"
+    FLAVOR_ENHANCER = "Geschmacksverstärker"
+    WAXED = "Gewachst"
+    PHOSPATES = "Phosphate"
+    SWEETENERS = "Süßungsmittel"
+    PHENYLALANINE = "Phenylaline"
+    COCOA_CONTAINING_GREASE = "Kakaohaltiges Fett"
+    GELATIN = "Gelatine"
+    ALCOHOL = "Alkohol"
+    PORK = "Schweinefleisch"
+    BEEF = "Rinderfleisch"
+    VEAL = "Kalbsfleisch"
+    WILD_MEAT = "Wildfleisch"
+    LAMB = "Lammfleisch"
+    GARLIC = "Knoblauch"
+    POULTRY = "Geflügel"
+    CEREAL = "Getreide"
+
+    # if an ingredient is a subclass of another ingredient,
+    # the superclass is added in the function lookup. E.g.: A hazelnut is also a shell fruit.
+    fmi_lookup: Dict[str, Set[Ingredient]] = {
+        "a": {GLUTEN},  # type: ignore
+        "aW": {WHEAT},  # type: ignore
+        "aR": {RYE},  # type: ignore
+        "aG": {BARLEY},  # type: ignore
+        "aH": {OAT},  # type: ignore
+        "aD": {SPELT},  # type: ignore
+        "aHy": {HYBRIDS},  # type: ignore
+        "b": {SHELLFISH},  # type: ignore
+        "c": {CHICKEN_EGGS},  # type: ignore
+        "d": {FISH},  # type: ignore
+        "e": {PEANUTS},  # type: ignore
+        "f": {SOY},  # type: ignore
+        "g": {MILK},  # type: ignore
+        "u": {LACTOSE},  # type: ignore
+        "h": {SHELL_FRUITS},  # type: ignore
+        "hMn": {ALMONDS},  # type: ignore
+        "hH": {HAZELNUTS},  # type: ignore
+        "hW": {WALNUTS},  # type: ignore
+        "hK": {CASHEWS},  # type: ignore
+        "hPe": {PECAN},  # type: ignore
+        "hPi": {PISTACHIOES},  # type: ignore
+        "hQ": {MACADAMIA},  # type: ignore
+        "i": {CELERY},  # type: ignore
+        "j": {MUSTARD},  # type: ignore
+        "k": {SESAME},  # type: ignore
+        "l": {SULFITES, SULPHURS},  # type: ignore
+        "m": {LUPIN},  # type: ignore
+        "n": {MOLLUSCS},  # type: ignore
+    }
+
+    studentenwerk_lookup: Dict[str, Set[Ingredient]] = {
+        "GQB": {BAVARIA},  # type: ignore
+        "MSC": {MSC},  # type: ignore
+        "1": {DYESTUFF},  # type: ignore
+        "2": {PRESERVATIVES},  # type: ignore
+        "3": {ANTIOXIDANTS},  # type: ignore
+        "4": {FLAVOR_ENHANCER},  # type: ignore
+        "5": {SULPHURS},  # type: ignore
+        "6": {DYESTUFF},  # type: ignore
+        "7": {WAXED},  # type: ignore
+        "8": {PHOSPATES},  # type: ignore
+        "9": {SWEETENERS},  # type: ignore
+        "10": {PHENYLALANINE},  # type: ignore
+        "11": {SWEETENERS},  # type: ignore
+        "13": {COCOA_CONTAINING_GREASE},  # type: ignore
+        "14": {GELATIN},  # type: ignore
+        "99": {ALCOHOL},  # type: ignore
+        # meatless is not an ingredient
+        "f": {},  # type: ignore
+        # vegan is not an ingredient
+        "v": {},  # type: ignore
```

Comment:
Even though meatless and vegan are no ingredients, I think it is still reasonable to save this information

Reply:
True. But maybe somewhere else?

Reply:
I think, it. would be better, to make the category Ingredients more general, such as Labels. 
For example GQB or MSC are also no real ingredients.

Alternatively one could also use the dish_type for this, but I think I would opt for the more general approach, of putting all information into the "renamed" Ingredients.